### ISSUE TITLE
ESLint errors not causing Webpasck to fail with non-zero exit code despite failOnError: true

### ISSUE THREAD

**Issue by iwsifatz** [webpack-contrib/eslint-webpack-plugin#274]
### Bug report

When using `eslint-webpack-plugin` with `failOnError: true`, ESLint errors are correctly detected and printed in the terminal. However, **Webpack does not terminate the build process with a non-zero exit code**. This results in the process continuing or being marked as successful, even when there are ESLint errors.

This is particularly problematic in CI/CD environments where we expect builds to fail on lint errors.

---

### Actual Behavior

Even with `failOnError: true` in the plugin options, Webpack displays ESLint errors as expected, but the CLI does **not exit with a non-zero status code**, nor does it terminate the process.

```
ERROR in × ESLintError: [eslint] 
  │ src/some/file.js
  │  5:10  error  'x' is assigned a value but never used  no-unused-vars
  │
  │ ✖ 1 problem (1 error, 0 warnings)


Webpack compiled with 1 error in 6.13 s  
```
💥 But process still exits with code 0.

---

### Expected Behavior

Webpack should:
- Respect `failOnError: true`
- Exit with a non-zero exit code (e.g., `1`)
- Stop further execution
---

### How Do We Reproduce?

1. Install and configure `eslint-webpack-plugin`:

```js
// webpack.config.js
const ESLintPlugin = require('eslint-webpack-plugin');

module.exports = {
  plugins: [
    new ESLintPlugin({
      failOnError: true,
      quiet: true,
    }),
  ],
};
```

2. Add a file with a lint error (e.g., `no-unused-vars`).
3. Run:

```bash
npx webpack--config webpack.config.js
```

4. Observe:
   - Lint error shown in terminal ✅
   - Process continues or exits with code 0 ❌

---

### Environment

```bash
npx webpack-cli info
```

- Webpack version: `@webpack/core@<version>`
- ESLint plugin: `eslint-webpack-plugin@<version>`
- Node.js: <version>
- OS: <OS info>
- ESLint: <version>
- CLI used: `webpack`, `npm`, etc.

---

### Notes

This seems to be either:
- A bug in `eslint-webpack-plugin`, where `failOnError` does not properly trigger process exit.
- A limitation/bug in Webpack's plugin system not respecting plugin errors in terms of exit code.

This behavior is different from `eslint-webpack-plugin`, which correctly causes the Webpack process to fail.

Please fix this behavior to ensure proper CI enforcement of ESLint errors. 🙏

**Comment by alexander-akait** [comment]
@ricardogobbosouza Can you look at this, thank you
